### PR TITLE
Assignment 6

### PR DESCRIPTION
**Pull request by rohitvishwakarma-sf**

### CODE REVIEW

File: client/src/services/UserServicesCrud.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+      if (response.status === 200) {
+        return "data changed";
+      }
+    } catch (error) {
+      console.log(error);
+    }
+  }
+}
+
+export const userServicesCrud = new UserServicesCrud();
```
Comment (sf-kansara) [inline client/src/services/UserServicesCrud.ts]:
```suggestion
```

File: client/src/controller/UserController.ts [hunk 2]
```diff
@@ -0,0 +1,229 @@
+import { Role, User } from "../model/User";
+import { userServicesCrud } from "../services/UserServicesCrud";
```
Comment (sf-kansara) [inline client/src/controller/UserController.ts]:
```suggestion
import { UserServicesCrud } from "../services/UserServicesCrud";
```

File: client/src/controller/UserController.ts [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+import { ClassConstants as CC } from "../model/classConstants";
+export class UserController {
+  users: User[] = [];
+  hostEle: HTMLDivElement;
+  tableBody?: HTMLDivElement;
+  constructor(hostId: string) {
+    this.users = [];
+    this.hostEle = document.getElementById(hostId)! as HTMLDivElement;
```
Comment (sf-kansara) [inline client/src/controller/UserController.ts:11]:
```suggestion
    this.hostEle = document.getElementById(hostId)! as HTMLDivElement;
    this.userServicesCrud = new UserServicesCrud();
```

File: client/src/controller/UserController.ts [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+  }
+
+  async load() {
+    this.users = await userServicesCrud.read();
```
Comment (sf-kansara) [inline client/src/controller/UserController.ts]:
```suggestion
    this.users = await this.userServicesCrud.read();
```

File: client/src/controller/UserController.ts [hunk 5]
```diff
@@ -0,0 +1,233 @@
...
+export class UserController {
+  users: User[] = [];
+  hostEle: HTMLDivElement;
+  tableBody?: HTMLDivElement;
+  userServicesCrud: UserServicesCrud;
+  constructor(hostId: string) {
+    this.users = [];
+    this.hostEle = document.getElementById(hostId)! as HTMLDivElement;
+    this.userServicesCrud = new UserServicesCrud();
+    this.userServicesCrud = new UserServicesCrud();
```
Comment (sf-kansara) [inline client/src/controller/UserController.ts]: Why twice?

File: client/src/controller/UserController.ts [hunk 6]
```diff
@@ -0,0 +1,233 @@
...
+      option.textContent = e;
+      if (user.role === e) option.selected = true;
+      else option.selected = false;
+      select.appendChild(option);
+    }
+    row.children[5].innerHTML = `<input value=${user.address}>`;
+    row.children[6].firstChild!.replaceWith(select);
+  }
+  create(user: User) {
+    this.users.push(user);
```
Comment (sf-kansara) [inline client/src/controller/UserController.ts]: Why is this not `this.userServicesCrud.create(user)`

Comment (rohitvishwakarma-sf) [reply client/src/controller/UserController.ts]: this is extra code, there is no use of it will remove it

**Review by sf-kansara — APPROVED** [review]

**Pull request merged**